<launch>    
  <arg name="config_file" default="$(find rwth_upper_body_detector)/config/config_Asus.inp" />
  <arg name="template_file" default="$(find rwth_upper_body_detector)/config/upper_temp_n.txt" />
  <arg name="queue_size" default="5" />
  <arg name="camera_namespace" default="/camera" />
  <arg name="upper_body_detections" default="/upper_body_detector/detections" />
  <arg name="detected_persons" default="/detected_persons"/>
  <arg name="detection_id_offset" default="0"/>
  <arg name="detection_id_increment" default="1"/>
  <arg name="depth_image" default="/depth/image_raw" />
  <arg name="rgb_image" default="/rgb/image_raw" />
  <arg name="camera_info_depth" default="/depth/camera_info" />
  <arg name="ground_plane" default="/spencer/sensors/rgbd_front_top/ground_plane"/>
  
  <group ns="/spencer/perception_internal/people_detection">
    <node pkg="turtlebot_follower_tracker" type="upper_body_detector" name="upper_body_detector" output="screen" respawn="true">
      <param name="config_file" value="$(arg config_file)" type="string"/>
      <param name="template_file" value="$(arg template_file)" type="string"/>
      <param name="queue_size" value="$(arg queue_size)" type="int"/>
      <param name="camera_namespace" value="$(arg camera_namespace)" type="string"/>
      <param name="depth_image" value="$(arg depth_image)" type="string"/>
      <param name="rgb_image" value="$(arg rgb_image)" type="string"/>
      <param name="camera_info_depth" value="$(arg camera_info_depth)" type="string"/>
      <param name="detected_persons" value="$(arg detected_persons)" type="string"/>
      <param name="upper_body_detections" value="$(arg upper_body_detections)" type="string"/>
      <param name="ground_plane" value="$(arg ground_plane)" type="string"/>
    </node>
  </group>
  
</launch>
